src/audio/drc/drc_math_hifi3.c

```diff
@@ -0,0 +1,263 @@
+	acc = AE_ADD32(acc, A3);
+	acc = mult(acc, x, 26, 30, 26);
+	acc = AE_ADD32(acc, A2);
+	acc = mult(acc, x, 26, 30, 26);
+	acc = AE_ADD32(acc, A1);
+	acc = mult(acc, x, 26, 30, 26);
+	acc = AE_ADD32(acc, A0);
+
+	tmp = mult(exp, LOG10_2, 25, 26, 26);
+	acc = AE_ADD32(acc, tmp);
+
+	return acc;
+}
+
+/*
+ * Input is Q6.26: max 32.0
+ * Output range ~ (-inf, 30.1030); regulated to Q11.21: (-1024.0, 1024.0)
+ */
+inline int32_t drc_lin2db_fixed(int32_t linear)
+{
+	ae_f32 log10_linear;
+
+	/* For negative or zero, just return a very small dB value. */
+	if (linear <= 0)
+		return -2097151999; /* Q_CONVERT_FLOAT(-1000.0f, 21); */
+
+	log10_linear = log10_fixed(linear); /* Q6.26 */
+	return mult(20 << 26, log10_linear, 26, 26, 21);
+}
+
+/*
+ * Input is Q6.26: max 32.0
+ * Output range ~ (-inf, 3.4657); regulated to Q6.26: (-32.0, 32.0)
+ */
+inline int32_t drc_log_fixed(int32_t x)
+{
+	/* const int32_t LOG10 = Q_CONVERT_FLOAT(2.3025850929940457f, 29); */
+	const ae_f32 LOG10 = 1236190976;
+	ae_f32 log10_x;
+
+	if (x <= 0)
+		return -2013265919; /* Q_CONVERT_FLOAT(-30.0f, 26); */
+
+	/* log(x) = log(10) * log10(x) */
+	log10_x = log10_fixed(x); /* Q6.26 */
+	return mult(LOG10, log10_x, 29, 26, 26);
+}
+
+/*
+ * Input is Q2.30; valid range: [-1.0, 1.0]
+ * Output range: [-1.0, 1.0]; regulated to Q2.30: (-2.0, 2.0)
+ */
+inline int32_t drc_asin_fixed(int32_t x)
+{
+	/* Coefficients obtained from:
+	 * If x <= 1/sqrt(2), then
+	 *   fpminimax(asin(x), [|1,3,5,7|], [|SG...|], [-1e-30;1/sqrt(2)], absolute)
+	 *   max err ~= 1.89936e-5
+	 * Else then
+	 *   fpminimax(asin(x), [|1,3,5,7|], [|SG...|], [1/sqrt(2);1], absolute)
+	 *   max err ~= 3.085226e-2
+	 */
+	/* const int32_t ONE_OVER_SQRT2 = Q_CONVERT_FLOAT(0.70710678118654752f, 30);
+	 * const int32_t TWO_OVER_PI = Q_CONVERT_FLOAT(0.63661977236758134f, 30);
+	 * const int32_t A7L = Q_CONVERT_FLOAT(0.1181826665997505187988281f, 30);
+	 * const int32_t A5L = Q_CONVERT_FLOAT(4.0224377065896987915039062e-2f, 30);
+	 * const int32_t A3L = Q_CONVERT_FLOAT(0.1721895635128021240234375f, 30);
+	 * const int32_t A1L = Q_CONVERT_FLOAT(0.99977016448974609375f, 30);
+	 * const int32_t A7H = Q_CONVERT_FLOAT(14.12774658203125f, 26);
+	 * const int32_t A5H = Q_CONVERT_FLOAT(-30.1692714691162109375f, 26);
+	 * const int32_t A3H = Q_CONVERT_FLOAT(21.4760608673095703125f, 26);
+	 * const int32_t A1H = Q_CONVERT_FLOAT(-3.894591808319091796875f, 26);
+	 */
+	const int32_t ONE_OVER_SQRT2 = 759250112; /* 1/sqrt(2) */
+	const ae_f32 TWO_OVER_PI = 683565248; /* 2/pi */
+	const ae_f32 A7L = 126897672;
+	const ae_f32 A5L = 43190596;
+	const ae_f32 A3L = 184887136;
+	const ae_f32 A1L = 1073495040;
+	const ae_f32 A7H = 948097024;
+	const ae_f32 A5H = -2024625535;
+	const ae_f32 A3H = 1441234048;
+	const ae_f32 A1H = -261361631;
+
+	ae_f32 in = x; /* Q2.30 */
+	ae_f32 in2 = mult(in, in, 30, 30, 30); /* Q2.30 */
+	ae_f32 A7, A5, A3, A1;
+	int32_t qc;
+	ae_f32 acc;
+
+	if (ABS((int32_t)in) <= ONE_OVER_SQRT2) {
```

Comment:
AE_ABS32S()?

Reply:
I think using standard abs() is simpler than AE_*:
1) AE_ABS32S() output type is ae_int32x2, we need to explicitly cast to ae_int32 then int.
2) ae_int32 requires AE_LE32 function to do comparison, and the output is ae_bool2, which also needs explicit cast to bool.